Services/CO.CDP.Organisation.WebApi/UseCase/GetApprovableOrganisationsUseCase.cs

```diff
@@ -0,0 +1,61 @@
+using CO.CDP.Organisation.WebApi.Model;
+using CO.CDP.OrganisationInformation;
+using CO.CDP.OrganisationInformation.Persistence;
+
+namespace CO.CDP.Organisation.WebApi.UseCase;
+public class GetApprovableOrganisationsUseCase(IOrganisationRepository organisationRepository)
+    : IUseCase<(string type, int limit, int skip), IEnumerable<Model.ApprovableOrganisation>>
+{
+    public async Task<IEnumerable<Model.ApprovableOrganisation>> Execute((string type, int limit, int skip) command)
+    {
+        var organisations = await organisationRepository.Get(command.type);
+
+        return organisations.Select(organisation => new ApprovableOrganisation
+            {
+                Id = organisation.Guid,
+                Name = organisation.Name,
+                Identifiers = GetIndentifiers(organisation),
+                Role = GetRole(organisation),
+                Email = GetEmail(organisation),
+                Ppon = GetPpon(organisation),
+                ApprovedOn = organisation.ApprovedOn,
+                ApprovedById = organisation.ApprovedBy?.Guid,
+                ApprovedByName = organisation.ApprovedBy?.FirstName + " " + organisation.ApprovedBy?.LastName,
+                ApprovedComment = organisation.ApprovedComment
+            })
+            .ToList();
+    }
+    private static string GetRole(OrganisationInformation.Persistence.Organisation organisation)
+    {
+        if (organisation.SupplierInfo != null)
+        {
+            return "supplier";
+        } else if (organisation.BuyerInfo != null)
+        {
+            return "buyer";
+        }
```

Comment:
This is a wrong way to identify buyers and suppliers.

`Organisation.Roles` tells you what roles organisation plays. It can be more than one. An organisation can be both a buyer and a supplier.

Reply:
We discussed this briefly and I still found a mismatch in the DB https://goacoglobal.slack.com/archives/C06M34UGL59/p1726239795145109